changeStreams.js

```diff
@@ -0,0 +1,142 @@
+
+        // OPTION THREE: Monitor new listings using the Stream API
+        // await monitorListingsUsingStreamAPI(client, 30000, pipeline);
+
+    } finally {
+        // Close the connection to the MongoDB cluster
+        await client.close();
+    }
+}
+
+main().catch(console.error);
+
+/**
+ * Close the given change stream after the given amount of time
+ * @param {*} timeInMs The amount of time in ms to monitor listings
+ * @param {*} changeStream The open change stream that should be closed
+ */
+function closeChangeStream(timeInMs = 60000, changeStream) {
+    return new Promise((resolve) => {
+        setTimeout(() => {
+            console.log("Closing the change stream");
+            changeStream.close();
+            resolve();
+        }, timeInMs)
+    })
+};
+
+/**
+ * Monitor listings in the listingsAndReviews collections for changes
+ * This function uses the on() function from the EventEmitter class to monitor changes
+ * @param {MongoClient} client A MongoClient that is connected to a cluster with the sample_airbnb database
+ * @param {Number} timeInMs The amount of time in ms to monitor listings
+ * @param {Object} pipeline An aggregation pipeline that determines which change events should be output to the console
+ */
+async function monitorListingsUsingEventEmitter(client, timeInMs = 60000, pipeline = []) {
+    const collection = client.db("sample_airbnb").collection("listingsAndReviews");
+
+    // See http://bit.ly/Node_watch for the watch() docs
+    const changeStream = collection.watch(pipeline);
+
+    // ChangeStream inherits from the Node Built-in Class EventEmitter (https://nodejs.org/dist/latest-v12.x/docs/api/events.html#events_class_eventemitter).
+    // We can use EventEmitter's on() to add a listener function that will be called whenever a change occurs in the change stream.
+    // See https://nodejs.org/dist/latest-v12.x/docs/api/events.html#events_emitter_on_eventname_listener for the on() docs.
+    changeStream.on('change', (next) => {
+        console.log(next);
+    });
+
+    await closeChangeStream(timeInMs, changeStream);
+}
+
+/**
+ * Monitor listings in the listingsAndReviews collections for changes
+ * This function uses the hasNext() function from the MongoDB Node.js Driver's ChangeStream class to monitor changes
+ * @param {MongoClient} client A MongoClient that is connected to a cluster with the sample_airbnb database
+ * @param {Number} timeInMs The amount of time in ms to monitor listings
+ * @param {Object} pipeline An aggregation pipeline that determines which change events should be output to the console
+ */
+async function monitorListingsUsingHasNext(client, timeInMs = 60000, pipeline = []) {
+    const collection = client.db("sample_airbnb").collection("listingsAndReviews");
+
+    // See http://bit.ly/Node_watch for the watch() docs
+    const changeStream = collection.watch(pipeline);
+
+    closeChangeStream(timeInMs, changeStream);
```

Comment:
suggest adding a comment here indicating why you are calling a method that looks like its closing the change stream _before_ iterating it